storage/gcloud/aio/storage/storage.py

```diff
@@ -368,7 +376,8 @@ async def _download(self, bucket: str, object_name: str, *,
                         params: Optional[Dict[str, str]] = None,
                         headers: Optional[Dict[str, str]] = None,
                         timeout: int = 10,
-                        session: Optional[Session] = None) -> bytes:
+                        session: Optional[Session] = None,
+                        get_content: bool = False) -> bytes:
```

Comment:
Looks like this breaks the expected return value here, since this method now returns a `aiohttp.StreamReader` when `get_content` is true and we're running in async mode -- I think this is a bit confusing since in the current method implementation, `response.content` refers to `bytes` since that code path is only hit by gcloud *rest* and the `requests` library uses that attribute to grab the raw non-stream bytes from the object. In this new code path, `response.content` refers to the attribute in `aiohttp`, which is completely different.

It may be easiest to just copy this out into a `_download_stream` method, for clarity sake -- see below comment for needing to support `gcloud-rest-storage` in this code anyway.

Reply:
Note that for a case like this, you may find it useful to make use of the `if BUILD_GCLOUD_REST` above for type hinting, for example you can do something like:
```python 
if BUILD_GCLOUD_REST:
    StreamResponse = io.BytesIO
else:
    StreamResponse = aiohttp.StreamReader

# SNIP
    async def _download_stream(..) -> StreamResponse
```

Reply:
> and we're running in async mode

Ohhhh, yeah I forgot this library has a synchronous mode as well. I was a little thrown off by [this try/except](https://github.com/talkiq/gcloud-aio/blob/a3bcfff89645b0717acc137414201c4550ab0c63/storage/gcloud/aio/storage/storage.py#L400), it looked like it was already breaking the return type. But I guess the Attribute or Type errors arise when you try to await a synchronous read, and therefore you know response.content will be `bytes` in that situation. Is that more or less it?

Anyway, _happy_ to break it out into another method. I almost did, and only didn't because I misunderstood that try/catch. I might add a comment to that try catch for posterity, if that's OK. :)

Reply:
More or less: the `await`s all get stripped out of the `gcloud-rest` code, but the `.read` attribute on `requests.Response` is (IIRC): not a method on old versions of requests (thus, raises a TypeError) and non-existent at all on newer versions (thus, raises an AttributeError).

Adding a comment mentioning what the except case is for sounds absolutely lovely, thank you!